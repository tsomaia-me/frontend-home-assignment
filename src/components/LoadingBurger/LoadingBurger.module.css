.root {
    width: 100%;
    height: 100%;
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;
    color: rgba(0, 0, 0, 0.5);
}

.animated {
    animation: loading 1s linear;
}

.root > span {
    font-size: 15rem;
}

@keyframes loading {
    from {
        opacity: 1;
        transform: scale(1);
    }
    to {
        opacity: 0;
        transform: scale(2);
    }
}
