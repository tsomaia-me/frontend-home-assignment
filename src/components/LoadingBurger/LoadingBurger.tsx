import React from 'react'

import styles from './LoadingBurger.module.css'

export interface LoadingBurgerProps {
  isAnimationEnabled: boolean
}

function LoadingBurger({ isAnimationEnabled }: LoadingBurgerProps) {
  return (
    <div className={`${styles.root}${isAnimationEnabled ? ' ' + styles.animated : ''}`}>
      <span className="material-icons">lunch_dining</span>
    </div>
  )
}

export default LoadingBurger
