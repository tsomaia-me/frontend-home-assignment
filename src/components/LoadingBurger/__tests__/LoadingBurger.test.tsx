import React from 'react'
import renderer from 'react-test-renderer'
import LoadingBurger from '../LoadingBurger'

test('renders LoadingBurger correctly', () => {
  const json = renderer.create(<LoadingBurger isAnimationEnabled={true}/>).toJSON()
  expect(json).toMatchSnapshot()
})
