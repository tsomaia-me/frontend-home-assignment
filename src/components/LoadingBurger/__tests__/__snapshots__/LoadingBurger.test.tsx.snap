// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders LoadingBurger correctly 1`] = `
<div
  className="root animated"
>
  <span
    className="material-icons"
  >
    lunch_dining
  </span>
</div>
`;
